pywr/recorders/recorders.py

```diff
@@ -198,14 +199,46 @@ def __init__(self, model, h5file, nodes=None, parameters=None, where='/', time='
         self.h5store = None
         self._arrays = {}
         self.nodes = nodes
-        self.parameters = parameters
         self.where = where
         self.time = time
         self.scenarios = scenarios
 
+        self.parameters = []
+        if parameters:
+            for parameter in parameters:
+                self._add_parameter(parameter)
+
         self._arrays = None
         self._time_table = None
 
+    def _add_parameter(self, parameter):
+        try:
+            where, param = parameter
+        except (TypeError, ValueError):
+            where = None
+            param = parameter
+        if isinstance(param, basestring):
+            param = load_parameter(self.model, param)
+        if not param.name:
```

Comment:
Can we add a check in here for parameter names containing "/". I've been bitten by that before. It creates random subgroups in the H5.